### PR TITLE
add: support to set more roles in web.xml

### PR DESCRIPTION
**Pull request by zlanyi**
Please check the following items before submitting a PR -- thank you!

Note that this project is released with a Contributor Code of Conduct.
By participating in this project you agree to abide by its terms.
[Contributor Code of Conduct](https://voxpupuli.org/coc/).

- [x] There is no existing PR that addresses this problem
- [x] Mentioned any existing issues in your commit so they get linked and
  closed once this PR gets merged, i.e: `Closes #1554` in the body of a commit
- [x] Followed the instructions in the [Contributing](CONTRIBUTING.md) document
- [x] Ran the unit/spec tests and ensured they still pass
- [x] Added tests to cover the new behaviour
- [x] Updated the documentation to match the changes
- [x] When possible, add an entry to the CHANGELOG file
- [ ] Squashed your PR down to a single commit. You may forego this if the PR
  tries to address multiple issues. Though we prefer one PR per feature/fix,
  sometimes that's not feasible. In that case ensure that a single feature/fix
  and associated tests and documentation is bundled up in one commit

Optional, but extra points:

- [ ] Added tests to ensure the old behaviour cannot accidentally be
  reintroduced

- - -

### CODE REVIEW

**Comment by zlanyi** [comment]
Error on Puppet 3 with Ruby 1.9.x:

> Gem::InstallError: rest-client requires Ruby version >= 2.0.0.
> Using puppetlabs_spec_helper 1.1.1
> An error occurred while installing rest-client (2.0.0), and Bundler cannot
> continue.
> Make sure that `gem install rest-client -v '2.0.0'` succeeds before bundling.
> The command "eval bundle install --without system_tests development --path=${BUNDLE_PATH:-vendor/bundle}" failed 3 times.

**Comment by zlanyi** [comment]
Hi,
any news here?:)

Cheers,
Zoltan

**Comment by bastelfreak** [comment]
Hi @zlanyi,

thanks for providing the PR: Two little things:
* Could you please rebase?
* Please check the used email address in your commits, github is unable to recognize and match them to your account

**Comment by zlanyi** [comment]
ah, i've changed my e-mail address, and now :D

**Comment by bastelfreak** [comment]
the email address is working now. Please check the git history, there are still merge conflicts.

**Comment by bastelfreak** [comment]
travis is green now, but again the email addresses are gone, except for one commit :D. I'm happy to merge this after your fixed the mail addresses.

**Comment by zlanyi** [comment]
ah, you are right :D

**Comment by bastelfreak** [comment]
could you check why there is a commit from @ltutar? Is this on purpose?

**Comment by zlanyi** [comment]
no idea, i rebased only from master and rewroted my email address.

**Comment by bastelfreak** [comment]
please rebase and kick it out of your history. You can join us on #voxpupuli on freenode if you need any help for slapping git.

**Comment by ltutar** [comment]
I think it is a mistake on my part. I was working on a different PR.

**Comment by ltutar** [comment]
Can you just reject my commit for this PR as admin?

**Comment by bastelfreak** [comment]
I can't. @zlanyi has to remove it during a rebase

**Comment by zlanyi** [comment]
I hope now its finite okey :)

File: manifests/config.pp [hunk 1]
```diff
@@ -51,6 +51,8 @@
   $truststore                  = $rundeck::truststore,
   $truststore_password         = $rundeck::truststore_password,
   $user                        = $rundeck::user,
+  $security_roles_array_enabled             = $rundeck::security_roles_array_enabled,
+  $security_roles_array                     = $rundeck::security_roles_array,
```
Comment (bastelfreak) [inline manifests/config.pp]: could you check this please? Maybe you've used hard tabs to align it, please try it with whitespace/soft tabs.

File: manifests/config/global/web.pp [hunk 2]
```diff
@@ -17,15 +17,28 @@
...
+# [*security_roles_array*]
+# Array value if you set the value 'security_roles_array_enabled' to true.
+#
 class rundeck::config::global::web (
-  $security_role   = $rundeck::params::security_role,
-  $session_timeout = $rundeck::params::session_timeout,
+  $security_role                            = $rundeck::params::security_role,
+  $session_timeout                          = $rundeck::params::session_timeout,
+  $security_roles_array_enabled             = $rundeck::params::security_roles_array_enabled,
+  $security_roles_array                     = $rundeck::params::security_roles_array,
```
Comment (bastelfreak) [inline manifests/config/global/web.pp]: the identation doesn't need to be that long

File: manifests/init.pp [hunk 3]
```diff
@@ -199,6 +215,8 @@
   $truststore                   = $rundeck::params::truststore,
   $truststore_password          = $rundeck::params::truststore_password,
   $user                         = $rundeck::params::user,
+  $security_roles_array_enabled              = $rundeck::params::security_roles_array_enabled,
+  $security_roles_array                      = $rundeck::params::security_roles_array,
```
Comment (bastelfreak) [inline manifests/init.pp]: please check identation

**Comment by bastelfreak** [comment]
the history is now looking fine, but the identation got screwed up in a few places. Could you please fix that?

**Comment by bastelfreak** [comment]
looks good!

**Comment by zlanyi** [comment]
Hi @bastelfreak 
thanks for your support :)

**Comment by bastelfreak** [comment]
sure sure, your welcome. I released 3.1.0 yesterday evening which includes your patch.

**Pull request merged**